model/import/AssemblerService.php

```diff
@@ -130,51 +138,98 @@ public function getExportDirectory()
+        foreach ($rdfIterator as $triple) {
+            if (!in_array($triple->predicate, $blacklist)) {
+                if (!isset($properties[$triple->predicate])) {
+                    $properties[$triple->predicate] = array();
                 }
+                $properties[$triple->predicate][] = $triple->object;
             }
         }
+
         return $properties;
     }
 
+    /**
+     * @param $folder
+     * @return FileIterator
+     *
+     * @throws AssemblyImportFailedException
+     */
+    protected function getRdfResourceIterator($folder)
+    {
+        $rdfPath = $folder.self::RDF_FILE;
+        if (!file_exists($rdfPath)) {
+            throw new AssemblyImportFailedException("Delivery rdf file {$rdfPath} does not exist");
+        }
+
+        return new FileIterator($rdfPath, 1);
+    }
+
     /**
      * @param core_kernel_classes_Class $deliveryClass
+     * @param $deliveryUri
+     *
      * @return core_kernel_classes_Resource
+     * @throws AssemblyImportFailedException
      */
-    protected function importDeliveryResource(core_kernel_classes_Class $deliveryClass)
+    protected function importDeliveryResource(core_kernel_classes_Class $deliveryClass, $deliveryUri)
     {
         $manifest       = $this->getDeliveryManifest();
         $label          = $manifest['label'];
         $dirs           = $manifest['dir'];
         $serviceCall    = \tao_models_classes_service_ServiceCall::fromString(base64_decode($manifest['runtime']));
-        $resultServer   = \taoResultServer_models_classes_ResultServerAuthoringService::singleton()->getDefaultResultServer();
 
-        $properties = $this->getAdditionalProperties($this->importFolder);
+        $properties = $this->getAdditionalProperties($this->getRdfResourceIterator($this->importFolder));
         $properties = array_merge($properties, array(
             OntologyRdfs::RDFS_LABEL                          => $label,
             DeliveryAssemblyService::PROPERTY_DELIVERY_DIRECTORY => array_keys($dirs),
             DeliveryAssemblyService::PROPERTY_DELIVERY_TIME      => time(),
             DeliveryAssemblyService::PROPERTY_DELIVERY_RUNTIME   => $serviceCall->toOntology(),
-            DeliveryContainerService::PROPERTY_RESULT_SERVER      => $resultServer
         ));
-        $delivery = $deliveryClass->createInstanceWithProperties($properties);
+
+        $delivery = $this->getResource($deliveryUri);
+        if ($delivery->exists()) {
+            throw new AssemblyImportFailedException("Delivery with this URI already exist: {$deliveryUri}");
+        }
+
+        $delivery->setType($deliveryClass);
+        $delivery->setPropertiesValues($properties);
 
         return $delivery;
     }
 
+    /**
+     * @param $useOriginalUri
+     * @return string
+     *
+     * @throws AssemblyImportFailedException
+     */
+    private function getDeliveryUri($useOriginalUri)
+    {
+        if ($useOriginalUri === false) {
```

Comment:
Currently if don't provide -uri param script already try use original uri because we have $useOriginalUri = NULL in this place. Is this the correct behavior? Could we have the possibility to run script with generating new uri ?

Reply:
Fixed